### PR TITLE
Create Models

### PR DESCRIPTION
**Pull request by SharifovDeveloper**

### CODE REVIEW

**Review by Nabiddinov — DISMISSED** [review]
goods

**Review by AzamatG17 — APPROVED** [review]

File: CheckDrive.Web/CheckDrive.Web/Models/Account.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        public string? Password { get; set; }
+        public string? PhoneNumber { get; set; }
+        public string? FirstName { get; set; }
+        public string? LastName { get; set; }
+        public DateTime Bithdate { get; set; }
+
+        public int RoleId { get; set; }
+        public Role? Role { get; set; }
+
+        public ICollection<Dispatcher>? Dispatchers { get; set; }
```
Comment (AzamatG17) [inline CheckDrive.Web/CheckDrive.Web/Models/Account.cs:16]: In here should be virtual(public virtual ICollection)

Comment (SharifovDeveloper) [reply CheckDrive.Web/CheckDrive.Web/Models/Account.cs:16]: Thanks for the your feedback, but what will change if I use it? Can you explain if possible

Comment (AzamatG17) [reply CheckDrive.Web/CheckDrive.Web/Models/Account.cs:16]: The virtual keyword enables polymorphism, allowing derived classes to override the property, while its absence indicates a non-overridable property.

Comment (SharifovDeveloper) [reply CheckDrive.Web/CheckDrive.Web/Models/Account.cs:16]: I know that. But I removed the virtual keyword from our old projects and didn't notice any change

**Review by diyorzz — DISMISSED** [review]

File: CheckDrive.Web/CheckDrive.Web/Models/Account.cs (hunk 1 above)
Comment (Mirazyzz) [reply CheckDrive.Web/CheckDrive.Web/Models/Account.cs:16]: https://stackoverflow.com/questions/25715474/why-navigation-properties-are-virtual-by-default-in-ef

File: CheckDrive.Web/CheckDrive.Web/Models/Account.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
+﻿namespace CheckDrive.Web.Models
+{
+    public class Account
+    {
+        public int Id { get; set; }
+        public string? Login { get; set; }
+        public string? Password { get; set; }
```
Comment (Mirazyzz) [inline CheckDrive.Web/CheckDrive.Web/Models/Account.cs:7]: Can user create an account without login and password?

File: CheckDrive.Web/CheckDrive.Web/Models/Account.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        public string? PhoneNumber { get; set; }
+        public string? FirstName { get; set; }
+        public string? LastName { get; set; }
+        public DateTime Bithdate { get; set; }
+
+        public int RoleId { get; set; }
+        public Role? Role { get; set; }
```
Comment (Mirazyzz) [inline CheckDrive.Web/CheckDrive.Web/Models/Account.cs:14]: If Role would be null what would be the value of a RoleId?

File: CheckDrive.Web/CheckDrive.Web/Models/Car.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
+﻿namespace CheckDrive.Web.Models
+{
+    public class Car
+    {
+        public int Id { get; set; }
+        public string? Model { get; set; }
+        public string? Color { get; set; }
+        public string? Number { get; set; }
```
Comment (Mirazyzz) [inline CheckDrive.Web/CheckDrive.Web/Models/Car.cs:8]: Can we create a car without model, color and number?

File: CheckDrive.Web/CheckDrive.Web/Models/Car.cs [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+        public double FuelConsumptionFor100Km { get; set; }
```
Comment (Mirazyzz) [inline CheckDrive.Web/CheckDrive.Web/Models/Car.cs:9]: We need a better name for this property

File: CheckDrive.Web/CheckDrive.Web/Models/Dispatcher.cs [hunk 6]
```diff
@@ -0,0 +1,11 @@
+﻿namespace CheckDrive.Web.Models
+{
+    public class Dispatcher
+    {
+        public int Id { get; set; }
+        public int AccountId { get; set; }
+        public Account? Account { get; set; }
```
Comment (Mirazyzz) [inline CheckDrive.Web/CheckDrive.Web/Models/Dispatcher.cs:7]: Marking account as nullable means that we can create a Dispatcher without an account, but AccountId is required which means we cannot create an Dispatcher without AccountId.

**Review by Mirazyzz — DISMISSED** [review]

**Review by AzamatG17 — APPROVED** [review]

**Review by AzamatG17 — APPROVED** [review]

**Review by diyorzz — APPROVED** [review]

**Pull request merged**